src/Catalyst.Core.Modules.Ledger/Web3EthApi.cs

```diff
@@ -22,25 +22,75 @@
     {
-        public Web3EthApi(IDeltaExecutor deltaExecutor, IStateReader stateReader, IStateRootResolver stateRootResolver, IDeltaResolver deltaResolver)
+        private readonly ITransactionRepository _receipts;
+        private readonly ITransactionReceivedEvent _transactionReceived;
+        public IHashProvider HashProvider { get; }
+        private readonly PeerId _peerId;
+
+        public Web3EthApi(IStateReader stateReader, IDeltaResolver deltaResolver, IDeltaCache deltaCache, IDeltaExecutor executor, IStorageProvider storageProvider, IStateProvider stateProvider, ITransactionRepository receipts, ITransactionReceivedEvent transactionReceived, IHashProvider hashProvider, IPeerSettings peerSettings)
         {
-            DeltaExecutor = deltaExecutor ?? throw new ArgumentNullException(nameof(deltaExecutor));
+            _receipts = receipts;
+            _transactionReceived = transactionReceived ?? throw new ArgumentNullException(nameof(transactionReceived));
+            HashProvider = hashProvider;
+            _peerId = peerSettings.PeerId;
+
             StateReader = stateReader ?? throw new ArgumentNullException(nameof(stateReader));
-            StateRootResolver = stateRootResolver ?? throw new ArgumentNullException(nameof(stateRootResolver));
             DeltaResolver = deltaResolver ?? throw new ArgumentNullException(nameof(deltaResolver));
+            DeltaCache = deltaCache ?? throw new ArgumentNullException(nameof(deltaCache));
+            Executor = executor ?? throw new ArgumentNullException(nameof(executor));
+            StorageProvider = storageProvider ?? throw new ArgumentNullException(nameof(storageProvider));
+            StateProvider = stateProvider ?? throw new ArgumentNullException(nameof(stateProvider));
         }
-        
-        public IDeltaExecutor DeltaExecutor { get; }
+
         public IStateReader StateReader { get; }
         public IDeltaResolver DeltaResolver { get; }
-        public IStateRootResolver StateRootResolver { get; }
+        public IDeltaCache DeltaCache { get; }
+
+        public IDeltaExecutor Executor { get; }
+        public IStorageProvider StorageProvider { get; }
+        public IStateProvider StateProvider { get; }
+
+        public Keccak SendTransaction(PublicEntry publicEntry)
+        {
+            TransactionBroadcast broadcast = new TransactionBroadcast
+            {
+                PublicEntry = publicEntry
+            };
+
+            _transactionReceived.OnTransactionReceived(broadcast.ToProtocolMessage(_peerId));
+
+            return new Keccak(HashProvider.ComputeMultiHash(broadcast).Digest);
+        }
+
+        public TransactionReceipt FindReceipt(Keccak transactionHash)
+        {
+            return _receipts.TryFind(transactionHash, out TransactionReceipt receipt) ? receipt : default;
+        }
+
+        public bool FindTransactionData(Keccak transactionHash, out Cid deltaHash, out Delta delta, out int index)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Consider refactoring this method in order to remove the need for this 'out' modifier.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4585091)